Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiCpuProfiler.inl

```diff
@@ -134,6 +136,116 @@ namespace AZ
+                ImGui::TableSetupColumn("MTPC (ms)");
+                ImGui::TableSetupColumn("Max (ms)");
+                ImGui::TableSetupColumn("Invocations");
+                ImGui::TableSetupColumn("Total (ms)");
+                ImGui::TableHeadersRow();
+                ImGui::TableNextColumn();
+
+                ImGuiTableSortSpecs* sortSpecs = ImGui::TableGetSortSpecs();
+                if (sortSpecs && sortSpecs->SpecsDirty)
+                {
+                    SortTable(sortSpecs);
+                }
+
+                // Draw all of the rows held in the GroupRegionMap
+                for (const auto* statistics : m_tableData)
+                {
+                    if (!m_timedRegionFilter.PassFilter(statistics->m_groupName.c_str())
+                        && !m_timedRegionFilter.PassFilter(statistics->m_regionName.c_str()))
+                    {
+                        continue;
+                    }
+
+                    ImGui::Text(statistics->m_groupName.c_str());
+                    const ImVec2 topLeftBound = ImGui::GetItemRectMin();
+                    ImGui::TableNextColumn();
+
+                    ImGui::Text(statistics->m_regionName.c_str());
+                    ImGui::TableNextColumn();
+
+                    ImGui::Text("%.2f", CpuProfilerImGuiHelper::TicksToMs(statistics->m_runningAverageTicks));
+                    ImGui::TableNextColumn();
+
+                    ImGui::Text("%.2f", CpuProfilerImGuiHelper::TicksToMs(statistics->m_maxTicks));
+                    ImGui::TableNextColumn();
+
+                    ImGui::Text("%ld", statistics->m_invocationsLastFrame);
+                    ImGui::TableNextColumn();
+
+                    ImGui::Text("%.2f", CpuProfilerImGuiHelper::TicksToMs(statistics->m_lastFrameTotalTicks));
+                    const ImVec2 botRightBound = ImGui::GetItemRectMax();
+                    ImGui::TableNextColumn();
+
+                    // NOTE: we are manually checking the bounds rather than using ImGui::IsItemHovered + Begin/EndGroup because
+                    // ImGui reports incorrect bounds when using Begin/End group in the Tables API.
+                    if (ImGui::IsWindowHovered() && ImGui::IsMouseHoveringRect(topLeftBound, botRightBound, false))
+                    {
+                        ImGui::BeginTooltip();
+                        ImGui::Text(statistics->GetExecutingThreadsLabel().c_str());
+                        ImGui::EndTooltip();
+                    }
+                }
+            }
+            ImGui::EndTable();
+        }
+
+        inline void ImGuiCpuProfiler::SortTable(ImGuiTableSortSpecs* sortSpecs)
+        {
+            const bool ascending = sortSpecs->Specs->SortDirection == ImGuiSortDirection_Ascending;
+            const ImS16 columnToSort = sortSpecs->Specs->ColumnIndex;
+                
+            switch (columnToSort)
+            {
+            case (0): // Sort by group name
```

Comment:
Anyone have ideas on how to make this less verbose? Maybe a predefined array of the comparison lambdas and then use `columnToSort` as an index into that array?

Reply:
Maybe it's not worth implementing but always a fun exercise on making code reusable. You can define a functor somewhere that takes in a function pointer to the member variable you want to compare
```
template <typename T>
class TableRowCompareFunctor {
    TableRowCompareFunctor(T memberPointer, bool isAscending) : m_memberPointer(functionPointer), m_isAscending(isAscending) {}
    TableRowCompareFunctor(const TableRow* lhs, const TableRow* rhs) {
              return m_isAscending ? lhs->*m_functionPointer < rhs->*m_functionPointer
    }

    T m_memberPointer;
    bool m_isAscending;
}
```

You'll still need the switch statement but it'll just be 
```
    ...
   case(0):
       AZStd::sort(m_tableData.begin(), m_tableData.end(), TableRowCompareFunctor(&TableRow::m_groupName, ascending)); 
```

Reply:
That is so clean! Always nice to start off the day learning something new, thanks! 😁